libraries/AP_Mount/AP_Mount_Xacti.cpp

```diff
@@ -622,43 +632,87 @@ void AP_Mount_Xacti::handle_param_save_response(AP_DroneCAN* ap_dronecan, const
     }
 }
 
+// get parameter name for a particular param enum value
+const char* AP_Mount_Xacti::get_param_name_str(Param param) const
+{
+    // check to avoid reading beyond end of array.  This should never happen
+    if ((uint8_t)param > ARRAY_SIZE(_param_names)) {
+        INTERNAL_ERROR(AP_InternalError::error_t::invalid_arg_or_result);
+        return nullptr;
```

Comment:
```suggestion
        return "";  // no caller will match this
```

... strcmp isn't save AFAIK to call with nullptr.  As in insta-crash

Reply:
Hi @peterbarker,

Thanks for the review.  I thought about passing back an empty string like you've suggested but I was worried about the life of that string.  Will it disappear once we return from the function?

BTW, I've added checks for nullptr so with the current code there's no chance of a nullptr being passed into strcmp.

Reply:
anyway, I've changed it to return an empty string

Reply:
> Thanks for the review. I thought about passing back an empty string like you've suggested but I was worried about the life of that string. Will it disappear once we return from the function?

I don't believe so; it's a constant string so should be valid after return.

> BTW, I've added checks for nullptr so with the current code there's no chance of a nullptr being passed into strcmp.

That works and is neater than passing back the empty string  May cost more flash.